b2share/modules/schemas/api.py

```diff
@@ -224,8 +272,14 @@ def create_version(self, json_schema):
             ).filter(
                 BlockSchemaVersionModel.block_schema == self.model.id
             ).one().last_version
+
             new_version = (last_version + 1 if last_version is not None
                            else 0)
+            if version_number and new_version < version_number:
+                raise InvalidSchemaVersionError()
```

Comment:
The exception should either be specific to "skipping version number" or it should receive a message explaining why the version number is invalid, i.e: "Provided version number should follow the last existing version number, which currently is <VERSION_NB>"